### PR TITLE
Forms stubs

### PR DESCRIPTION
**Pull request by zmitic**
As per suggestion of @VincentLanglet , I am making my first PR.

I am sorry for bad quality, codeception really gave me lots of problems but a hint here and there would be helpful to finalize it.

If the idea is not clear, I will write it down. It is something I have been using for about a year now, even had to annoy ``muglug`` for form listener accepting different event classes but it works.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "require": {
         "php": "^7.1 || ^8.0",
         "ext-simplexml": "*",
+        "symfony/form": "^4.0 || ^5.0",
```
Comment (VincentLanglet) [inline composer.json:15]: I think it should be in the require-dev instead

File: src/Plugin.php [hunk 2]
```diff
@@ -19,12 +19,24 @@
...
  * @psalm-suppress UnusedClass
  */
 class Plugin implements PluginEntryPointInterface
 {
+    /**
+     * @return string[]
+     */
+    protected function getFormStubs(): array
+    {
+        $forms = glob(__DIR__ . '/Stubs/Form/*.stubphp') ?: [];
```
Comment (seferov) [inline src/Plugin.php:34]: I am not sure about this subfolder. We can still organize form stubs under a separate subfolder but not at this level since it is all about symfony versions and forms can be a subject of version change. I think it can be organized like this

* stubs
    * common
         * form
         * other folders
    * 4
        * form 
        * other folders
    * 5
    ...

After changing structure to this there is no need to change this class.

File: composer.json (hunk 1 above)
Comment (seferov) [reply composer.json:15]: right, this must be in require-dev

**Review by seferov — CHANGES_REQUESTED** [review]
@zmitic thank you for the contribution.

File: src/Plugin.php (hunk 2 above)
Comment (zmitic) [reply src/Plugin.php:34]: I wanted to replicate structure that Symfony used, but can move them. Is it a problem to do that once everything else is done?
Just a note; there is no change between versions 4.4 (not sure about older) and 5.

File: composer.json (hunk 1 above)
Comment (zmitic) [reply composer.json]: Done.

File: src/Plugin.php [hunk 3]
```diff
@@ -25,6 +26,17 @@
...
+     * @return string[]
+     */
+    protected function getFormStubs(): array
+    {
+        $forms = glob(__DIR__.'/Stubs/common/forms/*.stubphp') ?: [];
+        $events = glob(__DIR__.'/Stubs/common/forms/Event/*.stubphp') ?: [];
+
+        return array_merge($forms, $events);
+    }
+
```
Comment (seferov) [inline src/Plugin.php:39]: this method (getFormStubs) is not needed.
Changing the `getCommonStubs` as following should do the work:

```php
    protected function getCommonStubs(): array
    {
        return glob(__DIR__.'/Stubs/common/{*,*/*}.stubphp', GLOB_BRACE) ?: [];
    }
```

File: src/Plugin.php [hunk 4]
```diff
@@ -78,7 +90,9 @@ public function __invoke(RegistrationInterface $api, SimpleXMLElement $config =
         $api->registerHooksFromClass(ContainerHandler::class);
 
         $stubs = array_merge(
-            $this->getCommonStubs(), $this->getStubsForMajorVersion(Kernel::MAJOR_VERSION)
+            $this->getCommonStubs(),
+            $this->getStubsForMajorVersion(Kernel::MAJOR_VERSION),
+            $this->getFormStubs()
```
Comment (seferov) [inline src/Plugin.php:95]: thus this change can be reverted

**Review by seferov — CHANGES_REQUESTED** [review]

File: src/Plugin.php (hunk 4 above)
Comment (zmitic) [reply src/Plugin.php]: Done, but I had to move stubs for events one folder up; it doesn't match Symfony folder structure anymore.

**Review by seferov — COMMENTED** [review]
@zmitic thank you. Could you please rebase your branch or merge with up-to-date master to pass the tests?

**Comment by zmitic** [comment]
@seferov Given that I totally messed up my local changes (never made PR before, always PHPStorm for git stuff), it will take few days till I figure how to untangle the mess I created.

Lesson learned: don't click options w/o understanding them first :smile:

**Comment by seferov** [comment]
@zmitic Thank you. Could you please merge with master again? Tests are failing

**Comment by zmitic** [comment]
@seferov 
>Thank you. Could you please merge with master again? Tests are failing

I did a merge but tests are still failing for 7.1. - 7.2.

Some help? Preferably ELI5?

File: src/Plugin.php [hunk 5]
```diff
@@ -85,7 +86,8 @@ public function __invoke(RegistrationInterface $api, SimpleXMLElement $config =
         $api->registerHooksFromClass(ContainerHandler::class);
 
         $stubs = array_merge(
-            $this->getCommonStubs(), $this->getStubsForMajorVersion(Kernel::MAJOR_VERSION)
+            $this->getCommonStubs(),
+            $this->getStubsForMajorVersion(Kernel::MAJOR_VERSION),
```
Comment (seferov) [inline src/Plugin.php:90]:
```suggestion
            $this->getStubsForMajorVersion(Kernel::MAJOR_VERSION)
```
tests for lower versions are failing because of this

Comment (zmitic) [reply src/Plugin.php]: **Thank you**, I honestly would not notice that. Fixed and commited.

File: src/Stubs/Forms/FormView.stubphp [hunk 6]
```diff
@@ -0,0 +1,158 @@
+<?php
```
Comment (seferov) [inline src/Stubs/Forms/FormView.stubphp:1]: `src/Stubs/Forms` directory and files inside should be deleted

**Review by seferov — CHANGES_REQUESTED** [review]

File: src/Stubs/common/forms/Form.stubphp [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+use Symfony\Component\Form\Exception\LogicException;
+use Symfony\Component\Form\Exception\OutOfBoundsException;
+use Symfony\Component\Form\Exception\RuntimeException;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+use Symfony\Component\Form\Exception\UnexpectedTypeException;
+use Symfony\Component\Form\Util\FormUtil;
+use Symfony\Component\Form\Util\InheritDataAwareIterator;
+use Symfony\Component\Form\Util\OrderedHashMap;
+use Symfony\Component\PropertyAccess\PropertyPath;
+use Symfony\Component\PropertyAccess\PropertyPathInterface;
```
[the remark this answers is not in the document]
Comment (zmitic) [reply src/Stubs/common/forms/Form.stubphp]: Thanks.

**Comment by seferov** [comment]
@zmitic thank you!
